code/game/objects/items/pet_carrier.dm

```diff
@@ -192,9 +206,51 @@
 /obj/item/pet_carrier/proc/remove_occupant(mob/living/occupant, turf/new_turf)
 	if(!(occupant in occupants) || !istype(occupant))
 		return
-	occupant.forceMove(new_turf ? new_turf : drop_location())
+	occupant.forceMove(new_turf ? new_turf : get_turf(src))
 	occupants -= occupant
 	occupant_weight -= occupant.mob_size
 	occupant.setDir(SOUTH)
 
+//bluespace jar, a reskin of the pet carrier that can fit people and smashes when thrown
+/obj/item/pet_carrier/bluespace
+	name = "bluespace jar"
+	desc = "A jar, that seems to be bigger on the inside, somehow allowing lifeforms to fit through its narrow entrance."
+	open = FALSE //starts closed so it looks better on menus
+	icon_state = "bluespace_jar"
+	item_state = "bluespace_jar"
+	lefthand_file = ""
+	righthand_file = ""
+	max_occupant_weight = MOB_SIZE_HUMAN //can fit people, like a bluespace bodybag!
+	load_time = 40 //loading things into a jar takes longer than a regular pet carrier
+	entrance_name = "lid"
+	w_class = WEIGHT_CLASS_NORMAL //it can fit in bags, like a bluespace bodybag!
+	throw_speed = 3
+	throw_range = 7
+	max_occupants = 1 //far less than a regular carrier or bluespace bodybag, because it can be thrown to release the contents
+	allows_hostiles = TRUE //can fit hostile creatures, with the move resist restrictions in place, this means they still cannot take things like legions/goliaths/etc regardless
+	has_lock_sprites = FALSE //jar doesn't show the regular lock overlay
+	custom_materials = list(/datum/material/glass = 1000, /datum/material/bluespace = 600)
+	escape_time = 10 //half the time of a bluespace bodybag
+
+/obj/item/pet_carrier/bluespace/update_icon_state()
+	if(open)
+		icon_state = "bluespace_jar_open"
+	else
+		icon_state = "bluespace_jar"
+
+/obj/item/pet_carrier/bluespace/throw_impact()
+	//delete the item upon impact, releasing the creature inside (this is handled by its deletion)
+	if(occupants.len)
+		src.loc.visible_message("<span class='warning'>The bluespace jar smashes, releasing [occupants[1]]!</span>")
+	qdel(src)
+	playsound(src, "shatter", 70, 1)
+	..()
+
+/obj/item/pet_carrier/bluespace/handle_internal_lifeform()
```

Comment:
do what cloning did and have a static gasmix instead of this
also consider supporting non humans/stuff that breathe other gas but idc

Reply:
instead decided to go and use traits so they're resistant while in the jar, because it's a far more efficient method for the pretty-much-same-effect